### PR TITLE
[bug]: Fix trial cleanup FK violation when deleting assistant config versions

### LINKED ISSUE #5175: Trial cleanup fails with FK violation when deleting assistant_config_versions
### Bug

Trial cleanup (Glific.Erase) fails continuously for orgs with assistants due to a foreign key violation:
```
\*\* (Glific.Erase.Error) Trial cleanup failed for org_id=368
Postgrex.Error: update or delete on table "assistant_config_versions" violates foreign key
constraint "assistants_active_config_version_id_fkey" on table "assistants"
Detail: Key (id)=(1984) is still referenced from table "assistants"
```

### Root Cause

In `lib/glific/erase.ex`, the cleanup runs in this order:
DELETE FROM assistant_config_versions ← fails here
DELETE FROM assistants ← active_config_version_id still points to above
The assistants table has active_config_version_id as a FK to assistant_config_versions. Deleting config versions before nulling out that reference causes the violation.

### Fix

Add an `UPDATE` before the delete to break the FK reference first:
```
UPDATE assistants SET active_config_version_id = NULL WHERE organization_id = ?
`DELETE FROM assistant_config_versions WHERE organization_id = ?
`DELETE FROM assistants WHERE organization_id = ?
```
### Impact
Every trial cleanup job for any org that has assistants fails continuously until fixed.

### PR DESCRIPTION
**Pull request by shijithkjayan**
## Summary
- Trial cleanup (`Erase.delete_organization_data/1`) was failing for any org with assistants due to a foreign key violation: `assistants.active_config_version_id` references `assistant_config_versions`, but deleting config versions while assistants still pointed to them was blocked by the `on_delete: :nothing` constraint
- Fixed by explicitly nulling `assistants.active_config_version_id` for the target org immediately before deleting `assistant_config_versions`, keeping the FK constraint strict in all other application flows
- Added a comprehensive test covering every table cleaned by `delete_organization_data/1`, including the FK violation scenario and edge cases (no assistants, assistant with no active config version)

Closes #5175

## Test plan
- [ ] Run `mix test test/glific/erase_test.exs` — all 18 tests pass
- [ ] Confirm trial cleanup no longer errors for orgs with assistants

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: priv/repo/migrations/20260603000000_fix_assistant_active_config_version_fk.exs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+defmodule Glific.Repo.Migrations.FixAssistantActiveConfigVersionFk do
+  use Ecto.Migration
+
+  def up do
+    drop_if_exists constraint(:assistants, :assistants_active_config_version_id_fkey)
+
+    alter table(:assistants) do
+      modify :active_config_version_id,
+             references(:assistant_config_versions, on_delete: :nilify_all),
+             null: true
```
[the remark this answers is not in the document]
Comment (shijithkjayan) [reply priv/repo/migrations/20260603000000_fix_assistant_active_config_version_fk.exs]: Keeping the migration approach. The  column is already nullable in the schema, and  accurately reflects the intended DB-level behaviour during hard deletes. The erase path is the only place that hard-deletes config versions, so the invariant holds in all normal application flows.

**Comment by shijithkjayan** [comment]
Switched approach per CodeRabbit's suggestion — removed the migration entirely (no schema change needed, FK stays `on_delete: :nothing`) and instead added an explicit `UPDATE assistants SET active_config_version_id = NULL WHERE organization_id = ?` in `Erase.delete_organization_data/1` immediately before deleting `assistant_config_versions`. This keeps the assistant invariant strict in all normal application flows.

**Review by AmishaBisht — APPROVED** [review]
